Regression in v1.19.0: AssertionError with module splitting on and optimizer off.

Scala version : 3.7.0-RC4
ScalaJS version : 1.19.0

Error happens with config ```ModuleSplitStyle.SmallModulesFor(...)``` or ```ModuleSplitStyle.ModuleSplitStyle.SmallestModules```.  
Does not happen with config `ModuleSplitStyle.FewestModules`

Error is thrown when running `fastLinkJS` but not `fullLinkJS`.

```
[info] Fast optimizing [...]
[error] java.util.concurrent.ExecutionException: Boxed Error
[error]         at scala.concurrent.impl.Promise$.resolver(Promise.scala:97)
[error]         at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:89)
[error]         at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:294)
[error]         at scala.concurrent.Promise.complete(Promise.scala:53)
[error]         at scala.concurrent.Promise.complete$(Promise.scala:52)
[error]         at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:197)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:42)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:74)
[error]         at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[error]         at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[error]         at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[error]         at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[error]         at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[error]         at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[error] Caused by: java.lang.AssertionError: assertion failed: java.util.function.Supplier.$$Lambda$ab10cd11e4c84f9dc7a29a581dc9938e4ecc6132 was not put in a module but has definitions
[error]         at scala.Predef$.assert(Predef.scala:223)
[error]         at org.scalajs.linker.frontend.modulesplitter.ModuleSplitter.$anonfun$assembleModules$2(ModuleSplitter.scala:76)
[error]         at scala.collection.immutable.List.foreach(List.scala:431)
[error]         at org.scalajs.linker.frontend.modulesplitter.ModuleSplitter.assembleModules(ModuleSplitter.scala:72)
[error]         at org.scalajs.linker.frontend.modulesplitter.ModuleSplitter.$anonfun$split$4(ModuleSplitter.scala:56)
[error]         at org.scalajs.logging.Logger.time(Logger.scala:42)
[error]         at org.scalajs.logging.Logger.time$(Logger.scala:40)
[error]         at org.scalajs.sbtplugin.Loggers$SbtLoggerWrapper.time(Loggers.scala:19)
[error]         at org.scalajs.linker.frontend.modulesplitter.ModuleSplitter.split(ModuleSplitter.scala:56)
[error]         at org.scalajs.linker.frontend.LinkerFrontendImpl.$anonfun$link$10(LinkerFrontendImpl.scala:87)
[error]         at org.scalajs.logging.Logger.time(Logger.scala:42)
[error]         at org.scalajs.logging.Logger.time$(Logger.scala:40)
[error]         at org.scalajs.sbtplugin.Loggers$SbtLoggerWrapper.time(Loggers.scala:19)
[error]         at org.scalajs.linker.frontend.LinkerFrontendImpl.$anonfun$link$9(LinkerFrontendImpl.scala:87)
[error]         at scala.util.Success.$anonfun$map$1(Try.scala:255)
[error]         at scala.util.Success.map(Try.scala:213)
[error]         at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[error]         at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:42)
[error]         at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:74)
[error]         at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[error]         at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[error]         at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[error]         at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[error]         at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[error]         at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[error] (ui / Compile / fastLinkJS) java.util.concurrent.ExecutionException: Boxed Error
```

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -264,6 +264,11 @@ def Tasks = [
         'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withModuleKind(ModuleKind.ESModule))' \
         $testSuite$v/test &&
     sbtretry ++$scala 'set Global/enableMinifyEverywhere := $testMinify' \
+        'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withModuleSplitStyle(ModuleSplitStyle.SmallestModules))' \
+        'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withModuleKind(ModuleKind.ESModule))' \
+        'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withOptimizer(false))' \
+        $testSuite$v/test &&
+    sbtretry ++$scala 'set Global/enableMinifyEverywhere := $testMinify' \
         'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withModuleSplitStyle(ModuleSplitStyle.SmallModulesFor(List("org.scalajs.testsuite"))))' \
         'set scalaJSLinkerConfig in $testSuite.v$v ~= (_.withModuleKind(ModuleKind.ESModule))' \
         $testSuite$v/test &&

--- a/linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala
+++ b/linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala
@@ -1488,6 +1488,7 @@ private class AnalyzerRun(config: CommonPhaseConfig, initial: Boolean,
         lookupOrSynthesizeClass(className, SyntheticClassKind.Lambda(descriptor)) { lambdaClassInfo =>
           lambdaClassInfo.instantiated()
           lambdaClassInfo.callMethodStatically(MemberNamespace.Constructor, ctorName)
+          moduleUnit.addStaticDependency(lambdaClassInfo.className)
         }
       }
     }

